components/CheckDocumentation/vacuum/XminHorizon.tsx

```diff
@@ -0,0 +1,342 @@
+    );
+    byStandby = heldBackBy.find((v: any) => v["type"] === "standby");
+    byPreparedXact = heldBackBy.find((v: any) => v["type"] === "prepared_xact");
+  }
+  const inApp = issue != null;
+  const causeTitle = inApp
+    ? heldBackBy.length > 1
+      ? "Identified Causes and Solutions"
+      : "Identified Cause and Solution"
+    : "Common Causes and Solutions";
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>VACUUM is potentially blocked by the xmin horizon.</p>
+      <p>
+        The xmin horizon tells you up to which point the vacuum process can
+        clean up dead rows. When this value is behind and not advancing, VACUUMs
+        will be blocked and will not be able to clean up dead rows.
+      </p>
+      <p>
+        When VACUUM is blocked and dead rows can't be cleaned, it can result to
+        table bloat and slow queries.
+      </p>
+      <h4>{causeTitle}</h4>
+      <ul>
+        <GuidanceByBackend inApp={inApp} heldBackInfo={byBackend} />
+        <GuidanceByReplicationSlot
+          inApp={inApp}
+          heldBackInfo={byReplicationSlot}
+          serverReplicationUrl={serverReplicationUrl}
+        />
+        <GuidanceByReplicationSlotCatalog
+          inApp={inApp}
+          heldBackInfo={byReplicationSlotCatalog}
+          serverReplicationUrl={serverReplicationUrl}
+        />
+        <GuidanceByStandby inApp={inApp} heldBackInfo={byStandby} />
+        <GuidanceByPreparedXact inApp={inApp} heldBackInfo={byPreparedXact} />
+      </ul>
+    </div>
+  );
+};
+
+const GuidanceByBackend: React.FunctionComponent<{
+  inApp: boolean;
+  heldBackInfo: number | null;
+}> = ({ inApp, heldBackInfo }) => {
+  if (inApp && !heldBackInfo) {
+    return null;
+  }
+
+  const CodeBlock = useCodeBlock();
+  return (
+    <li>
+      <h5>Long-running transactions</h5>
+      <p>
+        Long-running transactions may still need to access rows that could
+        otherwise be considered dead, so they can block cleanup.
+      </p>
+      <h6>
+        <b>Solution</b>
+      </h6>
+      {heldBackInfo && (
+        <p>
+          A long running transaction is holding back the xmin horizon at{" "}
+          <code>{heldBackInfo["xmin"]}</code>.
+        </p>
+      )}
+      <p>
+        You can find the transaction holding back the xmin horizon and its
+        connection's pid by running the following command:
+      </p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT pid, datname, usename, state, backend_xmin, backend_xid
+                  FROM pg_stat_activity
+                  WHERE backend_xmin IS NOT NULL OR backend_xid IS NOT NULL
+                  ORDER BY greatest(age(backend_xmin), age(backend_xid)) DESC;`}
+        />
+      </CodeBlock>
+      <p>You can cancel it by running either of commands:</p>
+      <CodeBlock>
+        <SQL
+          sql={`SELECT pg_cancel_backend('<query_pid>');
+                SELECT pg_terminate_backend('<query_pid>');`}
```

Comment:
I feel like giving users a choice here but not explaining the choice is not a good pattern to get into. An explanation of the difference between these two would be kind of a tangent, and I think `pg_cancel_backend` works pretty reliably in modern Postgres, so I think we should only mention that. (It would also be okay to only mention `pg_terminate_backend`; I do think mentioning both is potentially confusing.)